components/sidebar/sidebar_channel/index.js

```diff
@@ -30,18 +31,7 @@ function mapStateToProps(state, ownProps) {
         }
     }
 
-    let unreadMentions = 0;
-    let unreadMsgs = 0;
-    const m = ownProps.membership;
-
-    if (channel.type === 'D') {
-        unreadMentions = channel.total_msg_count - m.msg_count;
-    } else if (m.mention_count > 0) {
-        unreadMentions = m.mention_count;
-    }
-    if (m.notify_props && m.notify_props.mark_unread !== 'mention' && channel.total_msg_count - m.msg_count > 0) {
-        unreadMsgs = 1;
-    }
+    const unreads = ChannelStore.getUnreadCount(channel.id);
```

Comment:
Would it be possible to use a selector from redux for this? The RN app must already be doing this right, or is there special behaviour in the webapp that we need?

Reply:
The RN app already does this, I'll be submitting a proper fix along side the unread section PR

Reply:
Sounds good

Reply:
Take into consideration that other parts of the webapp are doing staff directly into the store. The ChannelStore emits redux events to update the counts of messages, but at the same time it is counting the messages in their own memory. What I mean is that is not simply getting a reducer that calculate correctly the counter, you also need to avoid that the rest of the frontend application mess up with the store. An example of this is the incrementMessage method of the ChannelStore.